utils/apiObjects.ts

```diff
@@ -0,0 +1,151 @@
+import { hexToDecimal, stringToFelt } from "./feltService";
+import { formatHexString } from "./stringService";
+
+const STARKNET = formatHexString(stringToFelt("starknet"));
+const DISCORD = formatHexString(stringToFelt("discord"));
+const TWITTER = formatHexString(stringToFelt("twitter"));
+const GITHUB = formatHexString(stringToFelt("github"));
+const PROOF_OF_PERSONHOOD = formatHexString(
+  stringToFelt("proof_of_personhood")
+);
+
+export class Identity {
+  private data: IdentityData;
+  private userDataMap: Map<string, string>;
+  private verifierDataMap: Map<string, string>;
+
+  constructor(identityData: IdentityData) {
+    this.data = identityData;
+
+    this.userDataMap = new Map();
+    identityData.user_data.forEach((ud) => {
+      this.userDataMap.set(ud.field, ud.data);
+    });
+
+    this.verifierDataMap = new Map();
+    identityData.verifier_data.forEach((vd) => {
+      this.verifierDataMap.set(`${vd.verifier}_${vd.field}`, vd.data);
+    });
+  }
+
+  getData(): IdentityData {
```

Comment:
why do you use only getters here ?

Reply:
for encapsulation

Reply:
I rewrote getters with js get syntax to preserve encapsulation but keep a property syntax. See https://github.com/starknet-id/app.starknet.id/pull/580/commits/9ad4f9c26526ea3c50e7d654f1ccb9b406eebc84